lesson3/Lesson3HW/app/src/main/java/com/margaret/lesson3hw/MainActivity.java

```diff
@@ -32,25 +32,35 @@ public void onClick(View view) {
         });
 
         //set up fragment manager stuff
-        final Fragment settingsFragment = new SettingsFragment();
         Fragment defaultFragment = new MainActivityFragment();
-        FragmentManager fragmentManager = getSupportFragmentManager();
-        final FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
+        final FragmentManager fragmentManager = getSupportFragmentManager();
+        FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
```

Comment:
So I figured out what was happening here, not sure why it worked on your old phone before...

You were using the same `FragmentTransaction` each time you needed to switch fragments, which won't work.  You can use the same FragmentManager, as I did here, but make a new transaction each time by calling `beginTransaction()` on your manager.